lib/Cake/I18n/I18n.php

```diff
@@ -435,7 +435,21 @@ protected function _bindTextDomain($domain) {
 				if ($translations === false && is_file($file . '.po')) {
 					$translations = static::loadPo($file . '.po');
 				}
+				
+				if ($translations !== false) {
+
+					# PHP 8.1 fix
+					# Deprecated: Automatic conversion of false to array is deprecated
+					if(!is_array($this->_domains[$domain][$this->_lang])) {
+						$this->_domains[$domain][$this->_lang] = [];
+					}
 
+					$this->_domains[$domain][$this->_lang][$this->category] = $translations;
+					$this->_noLocale = false;
+					break 2;
+					
+				}
+				
```

Comment:
Hey @kallealberg Thanks for your contribution! :blush: 

I'm wondering about one thing. Shouldn't we remove the `if` statement from below? Isn't it redundant? 

I see that the same logic is already covered in the code added above + extra check to make sure array is properly initialized.